CRM/Core/Payment/OmnipayMultiProcessor.php

```diff
@@ -839,7 +839,7 @@ protected function redirectOrExit($outcome, $response) {
         if (empty($redirectUrl)) {
           $redirectUrl = $this->getReturnSuccessUrl(CRM_Utils_Request::retrieve('qfKey', 'String'));
         }
-        if (method_exists($response, 'confirm')) {
+        if ($redirectUrl && method_exists($response, 'confirm')) {
```

Comment:
@eileenmcnaughton The !$redirectUrl was added by you recently, but I'm not sure I understand why.  In sagepay case at least it needs to have a redirectUrl in order to call those functions as that redirect URL is passed back to sagepay via the IPN handler.  Uniquely(?) for sagepay the confirm/error/invalid methods do not return as it is up to the sagepay website to initiate the redirect.